### PR TITLE
opensc: 0.25.1 -> 0.26.0

### PR DESCRIPTION
**Pull request by michaeladler**
**Security fixes**:

https://github.com/advisories/GHSA-3q68-hm47-94vg: Usage of uninitialized values in libopensc and pkcs15init (https://github.com/OpenSC/OpenSC/pull/3225)
    https://github.com/advisories/GHSA-2mjg-798r-mxwh: Uninitialized values after incorrect check or usage of APDU response values in libopensc (https://github.com/OpenSC/OpenSC/pull/3225)
    https://github.com/advisories/GHSA-cf2w-h975-2fpg: Uninitialized values after incorrect or missing checking return values of functions in libopensc (https://github.com/OpenSC/OpenSC/pull/3225)
    https://github.com/advisories/GHSA-f2v6-mw6x-qmwc: Uninitialized values after incorrect or missing checking return values of functions in pkcs15init (https://github.com/OpenSC/OpenSC/pull/3225)
    https://github.com/advisories/GHSA-9vxw-3j77-cj78: Incorrect handling length of buffers or files in libopensc (https://github.com/OpenSC/OpenSC/pull/3225)
    https://github.com/advisories/GHSA-9c2g-6v5v-57qg: Incorrect handling of the length of buffers or files in pkcs15init (https://github.com/OpenSC/OpenSC/pull/3225)
    https://github.com/advisories/GHSA-mgc5-p43f-72pc: Heap buffer overflow in OpenPGP driver when generating key (https://github.com/OpenSC/OpenSC/pull/3219)

## Things done

Also: move to pkgs/by-name, format, add updateScript

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

**Comment by Pandapip1** [comment]
Merge conflict due to https://github.com/NixOS/nixpkgs/pull/354531.

**Comment by michaeladler** [comment]
Rebased

**Review by FliegendeWurst — COMMENTED** [review]
Darwin build fails
```
checking for winscard.h... no
configure: error: winscard.h is required for pcsc
```

**Comment by pbsds** [comment]
> Darwin build fails
> 
> ```
> checking for winscard.h... no
> configure: error: winscard.h is required for pcsc
> ```

preexisting https://cache.nixos.org/log/0y56cxr064jp7dhllm5zfyahzr79kp14-opensc-0.25.1.drv

**Comment by numinit** [comment]
We probably should backport to 24.11 too because of branchoff.

File: pkgs/by-name/op/opensc/package.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        "${darwin.apple_sdk.frameworks.PCSC}/Library/Frameworks/PCSC.framework/PCSC"
+      else
+        "${lib.getLib pcsclite}/lib/libpcsclite${stdenv.hostPlatform.extensions.sharedLibrary}"
+    }"
+    (lib.optionalString (
+      stdenv.hostPlatform != stdenv.buildPlatform
+    ) "XSLTPROC=${buildPackages.libxslt}/bin/xsltproc")
+  ];
+
+  PCSC_CFLAGS = lib.optionalString withApplePCSC "-I${darwin.apple_sdk.frameworks.PCSC}/Library/Frameworks/PCSC.framework/Headers";
```
Comment (numinit) [inline pkgs/by-name/op/opensc/package.nix:76]: I think we still need to add `"${pcsclite}/include"` here

Comment (michaeladler) [reply pkgs/by-name/op/opensc/package.nix:76]: Yes adding `${lib.getDev pcsclite}/include/PCSC` fixes the build. I will submit a separate MR.

Comment (numinit) [reply pkgs/by-name/op/opensc/package.nix:76]: Thank you!

**Pull request merged**